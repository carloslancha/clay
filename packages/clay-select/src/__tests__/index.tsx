/**
 * Â© 2019 Liferay, Inc. <https://liferay.com>
 *
 * SPDX-License-Identifier: BSD-3-Clause
 */

import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import ClaySelect from '..';

const options = [
	{
		label: 'Option 1',
		value: '1'
	},
	{
		label: 'Option 2',
		value: '2'
	},
	{
		label: 'Option 3',
		value: '3'
	}
];

describe('Rendering', () => {
	it('with required attributes', () => {
		const testRenderer = TestRenderer.create(
			<ClaySelect 
				label='Select Label'
				options={options}
			/>
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});

	it('with container properties', () => {
		const testRenderer = TestRenderer.create(
			<ClaySelect 
				containerProps={{
					className:"customContainerClassName"
				}}
				label='Select Label'
				options={options}
			/>
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});

	it('with id', () => {
		const testRenderer = TestRenderer.create(
			<ClaySelect 
				id="mySelectId"
				label='Select Label'
				options={options}
			/>
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});

	it('with name', () => {
		const testRenderer = TestRenderer.create(
			<ClaySelect 
				label='Select Label'
				name="mySelectName"
				options={options}
			/>
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});

	it('disabled', () => {
		const testRenderer = TestRenderer.create(
			<ClaySelect 
				disabled
				label='Select Label'
				options={options}
			/>
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});

	it('with multiple options allowed', () => {
		const testRenderer = TestRenderer.create(
			<ClaySelect 
				label='Select Label'
				multiple
				options={options}
			/>
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});

	it('inline', () => {
		const testRenderer = TestRenderer.create(
			<ClaySelect 
				inline={true}
				label='Select Label'
				options={options}
			/>
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});
});
