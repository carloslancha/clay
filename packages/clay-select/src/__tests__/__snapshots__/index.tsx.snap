// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering disabled 1`] = `
<div
  className="form-group"
>
  <label>
    Select Label
  </label>
  <select
    aria-label="Select Label"
    className="form-control"
    disabled={true}
  >
    <option
      value="1"
    >
      Option 1
    </option>
    <option
      value="2"
    >
      Option 2
    </option>
    <option
      value="3"
    >
      Option 3
    </option>
  </select>
</div>
`;

exports[`Rendering inline 1`] = `
<div
  className="form-group-item"
>
  <label>
    Select Label
  </label>
  <select
    aria-label="Select Label"
    className="form-control"
  >
    <option
      value="1"
    >
      Option 1
    </option>
    <option
      value="2"
    >
      Option 2
    </option>
    <option
      value="3"
    >
      Option 3
    </option>
  </select>
</div>
`;

exports[`Rendering with container properties 1`] = `
<div
  className="customContainerClassName form-group"
>
  <label>
    Select Label
  </label>
  <select
    aria-label="Select Label"
    className="form-control"
  >
    <option
      value="1"
    >
      Option 1
    </option>
    <option
      value="2"
    >
      Option 2
    </option>
    <option
      value="3"
    >
      Option 3
    </option>
  </select>
</div>
`;

exports[`Rendering with id 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="mySelectId"
  >
    Select Label
  </label>
  <select
    aria-label="Select Label"
    className="form-control"
    id="mySelectId"
  >
    <option
      value="1"
    >
      Option 1
    </option>
    <option
      value="2"
    >
      Option 2
    </option>
    <option
      value="3"
    >
      Option 3
    </option>
  </select>
</div>
`;

exports[`Rendering with multiple options allowed 1`] = `
<div
  className="form-group"
>
  <label>
    Select Label
  </label>
  <select
    aria-label="Select Label"
    className="form-control"
    multiple={true}
  >
    <option
      value="1"
    >
      Option 1
    </option>
    <option
      value="2"
    >
      Option 2
    </option>
    <option
      value="3"
    >
      Option 3
    </option>
  </select>
</div>
`;

exports[`Rendering with name 1`] = `
<div
  className="form-group"
>
  <label>
    Select Label
  </label>
  <select
    aria-label="Select Label"
    className="form-control"
    name="mySelectName"
  >
    <option
      value="1"
    >
      Option 1
    </option>
    <option
      value="2"
    >
      Option 2
    </option>
    <option
      value="3"
    >
      Option 3
    </option>
  </select>
</div>
`;

exports[`Rendering with required attributes 1`] = `
<div
  className="form-group"
>
  <label>
    Select Label
  </label>
  <select
    aria-label="Select Label"
    className="form-control"
  >
    <option
      value="1"
    >
      Option 1
    </option>
    <option
      value="2"
    >
      Option 2
    </option>
    <option
      value="3"
    >
      Option 3
    </option>
  </select>
</div>
`;
